### PR TITLE
Hotfix/base template update

### PR DESCRIPTION
**Pull request by VirginiaDooley**
Closes https://trello.com/c/oyrIDWEA/3048-bump-dcutils-to-version-21-in-all-projects and https://trello.com/c/EGiFT42z/3042-edit-minimum-versions-for-dependabot

Dependent on https://github.com/DemocracyClub/dc_django_utils/pull/42

- Match `dc_utils==2.0.2` base template structure
- Upgrade `Django>=4.1.2` and `Python==3.8.12` with necessary dependency upgrades
- Introduces a [temporary Django 4.0 compatibility fix ](https://github.com/matthewn/django-typogrify/tree/django40)for `typogrify_tags`. 
- Replaces `markdown_deux` (has deprecated imports with `Django>4.0` and is not actively maintained) with `Markdown`.

### CODE REVIEW

File: democracy_club/templates/base.html [hunk 1]
```diff
@@ -5,128 +5,65 @@
...
 {% endblock extra_site_css %}
 
-{% block body_base %}
-    <body class="ds-width-full">
-        <div class="ds-page">
-            <p><a class="ds-skip-link" href="#main">skip to content</a></p>
-            {% block header %}
-                {% block site_menu %}
-                    <header class="ds-header">
-                        <a class="ds-logo" href="/">
```
Comment (VirginiaDooley) [inline democracy_club/templates/base.html:47]: Currently, we have `SITE_WIDTH_LOGO` set in `dc_utils` context processors to `"300"`. I've tried to override this locally in `base.py` to fix the logo spacing issue shown here, but it remains at 300:

Screen Shot 2022-10-21 at 5 07 23 PM

**Comment by VirginiaDooley** [comment]
[Edit minimum versions for dependabot](https://trello.com/c/EGiFT42z/3042-edit-minimum-versions-for-dependabot)

**Comment by VirginiaDooley** [comment]
[Bump dc_utils to version 2.1 in all projects](https://trello.com/c/oyrIDWEA/3048-bump-dcutils-to-version-21-in-all-projects)

**Comment by Bekabyx** [comment]
All looks good on my local version now, thanks for fixing the pipfile mismatch 😁

**Review by Bekabyx — APPROVED** [review]

File: democracy_club/templates/base.html (hunk 1 above)
Comment (VirginiaDooley) [reply democracy_club/templates/base.html:47]: Ignore this; it turns out this is unrelated to settings. This spacing is on the `img` tag.

**Review by symroe — APPROVED** [review]

**Pull request merged**